crates/matrix-sdk/src/sliding_sync/view.rs

```diff
@@ -631,10 +631,17 @@ impl SlidingSyncViewRequestGenerator {
         limit: Option<u32>,
     ) -> v4::SyncRequestList {
         let calc_end = start + batch_size;
-        let end = match limit {
+
+        let mut end = match limit {
             Some(l) => std::cmp::min(l, calc_end),
             _ => calc_end,
         };
+
+        end = match self.view.rooms_count() {
+            Some(total_room_count) => std::cmp::min(end, std::cmp::max(0, total_room_count - 1)),
```

Comment:
There's `clamp` for this: https://doc.rust-lang.org/stable/std/cmp/trait.Ord.html#method.clamp